src/model/Params.ts

```diff
@@ -0,0 +1,246 @@
+  electricitySolar: 51.42 as TWh,
+  electricityWind: 131.85 as TWh,
+  electricityWindOnshoreMaxNew: 6.0 as TWh,
+  electricityWater: 14.99 as TWh,
+  electricityHardCoal: 35.46 as TWh,
+  electricityBrownCoal: 82.13 as TWh,
+  electricityBiomass: 43.19 as TWh,
+  electricityNuclear: 60.91 as TWh,
+
+  buildingsPrivateDemand: 544 as TWh,
+  buildingsIndustryDemand: 226 as TWh,
+  buildingsSourceBio: 130 as TWh,
+  buildingsSourceOil: 219 as TWh,
+  buildingsSourceTele: 58 as TWh,
+
+  co2emissionsAgriculture: 67.936, uba2020DeutscheTreibhausgasEmissionen
+  co2emissionsOthers: 9.243,uba2020DeutscheTreibhausgasEmissionen
+
+
+
+  get electricityGas() {
+    return (
+      this.electricityDemand -
+      this.electricitySolar -
+      this.electricityWind -
+      this.electricityWater -
+      this.electricityHardCoal -
+      this.electricityBrownCoal -
+      this.electricityBiomass -
+      this.electricityNuclear
+    )
+  },
+
+  get co2emissionsEnergy() {
+    // should sum up to 220 in 2020
+    // factors from https://www.rensmart.com/Calculators/KWH-to-CO2 and @thomas-olszamowski
+    return (
+      this.electricityGas * 0.399 +
+      this.electricitySolar * 0.058 +
+      this.electricityWind * 0.005 +
+      this.electricityWater * 0.02 +
+      this.electricityHardCoal * 0.835 +
+      this.electricityBrownCoal * 1.137 +
+      this.electricityBiomass * 0 + // TODO find correct factor (no source found)
+      this.electricityNuclear * 0.005
+    )
+  },
+
+  get passengerTransportUsage(): MioPsgrKm {
+    return (
+      this.carUsage + this.publicLocalUsage + this.publicNationalUsage + this.airDomesticUsage + this.airIntlUsage
+    )
+  },
+
+
+  get buildingsDemand(): TWh {
+    return this.buildingsPrivateDemand + this.buildingsIndustryDemand
+  },
+
+  get buildingsSourceGas(): TWh {
+    return this.buildingsDemand - (this.buildingsSourceBio + this.buildingsSourceOil + this.buildingsSourceTele)
+  },
+
+  get co2emissionsBuildings(): MioTons {
+    // Should initially equal 123.461 MioTons (https://www.umweltbundesamt.de/sites/default/files/medien/361/dokumente/2021_03_10_trendtabellen_thg_nach_sektoren_v1.0.xlsx sheet "THG" row 2019)
+    return (
+      this.buildingsSourceBio * 0 +
+      this.buildingsSourceGas * 0.247 + // https://www.polarstern-energie.de/magazin/artikel/heizen-co2-vergleich-von-brennstoffen/#c7085
+      this.buildingsSourceOil * 0.318 + // https://www.polarstern-energie.de/magazin/artikel/heizen-co2-vergleich-von-brennstoffen/#c7085
+      this.buildingsSourceTele * 0.16
+    ) // from https://www.klimaneutral-handeln.de/php/kompens-berechnen.php#rechner
+  },
+
+  get co2emissions(): number {
+    return (
+      this.co2emissionsEnergy +
+      this.co2emissionsIndustry +
+      this.co2emissionsMobility +
+      this.co2emissionsBuildings +
+      this.co2emissionsAgriculture +
+      this.co2emissionsOthers
+    )
+  },
+}
+*/
+}
+
+export type ParamDefinitions = typeof paramDefinitions
```

Comment:
... bekomme dann hier aber 
```typescript
type ParamDefinitions = Record<string, ParamDefinition>
```

Reply:
Ist doch auch nicht falsch? Vermutlich willst du hier statt des generischen Strings die echten Keys, die auch definiert wurden haben?

Du könntest dann hier das schreiben:
`type ParamDefinitions = Record<keyof typeof params, ParamDefinition>`

Reply:
Wüsste nicht, wie man das so ganz ohne Wiederholung schaffen kann, aber eine nur geringe Wiederholung wäre:

```
type ParamKey = "a" | "b" | "c"
export const paramDefinitions: Record<ParamKey, ParamDefinition> = { ... }
type ParamDefinitions = Record<ParamKey, ParamDefinition>
```

Dann werden zumindest falsche ParamDefinitions bereits bei der Definition erkannt und du hast am Ende einen sehr konkreten `ParamDefinitions` Typ, in dem nicht einfach irgendwas geht, sondern nur genau die definierten Attribute.

Reply:
`Record<keyof typeof paramDefinitions, ParamDefinition>` ist genau, was ich brauche... Die `ParamKey` Lösung wäre nur nötig, wenn wir damit besser kontrollieren wollen, welche es geben soll. Das soll aber in `paramDefinitions` zentral und an einziger Stelle festgelegt werden - mit allen zugehörigen Infos dabei. Super! Danke!

Reply:
Geht leider doch nicht:
```typescript
export const paramDefinitions: Record<string, ParamDefinition> = {...}
export type ParamDefinitions = Record<keyof typeof paramDefinitions, ParamDefinition>
```
liefert leider einfach nur
```typescript
type ParamDefinitions = {
    [x: string]: ParamDefinition;
}
```

Reply:
Ich versuche es jetzt analog zu den Laws und Events: Jede ParamDefinition einer einzelnen Variablen zuordnen und dann `paramDefinitions` daraus zusammenbauen.

Reply:
Ich denke, wenn du die Konstante `paramDefinitions` explizit mit dem Typ `Record<string, ParamDefinition>` definierst, dann kann eigentlich auch nur das raus kommen. Konkrete Werte kannst du eigentlich nur auf zwei Wegen in den Typ rein bekommen, entweder du gibst sie explizit an (über ein enum oder mit Literalen), oder du lässt Typescript das selbst ermitteln ("Type inference"), dann musst du den Typ weg lassen, hast aber den Nachteil, dass du im Prinzip innerhalb der Konstante angeben kannst, was du willst (und eben auch falsches).

Reply:
...irgendwie ist es ein ähnliches Problem, wie bei `laws/index.js`.. Danke!